### DISCUSSION TITLE
Generating plots in a loop

### DISCUSSION THREAD

**Discussion by lucas-a-meyer** [quarto-dev/quarto-cli#936] [Q&A] [answered]
I want to display several time series in a loop after their titles.

What I am trying to do is to put a title first in Markdown, and then retrieve a series and plot it using the `DataFrame.plot()` method.

What's happening is that when I render, I first get all the titles and then I get all the figures (title, title title, image, image, image). I wanted to get title, image, title, image, title, image, etc.

```
# Create list of geographies of interest
geographies_dict = [
  {'geo_name':'United States', 'geo_code':'USA'},
  {'geo_name':'Japan', 'geo_code':'JPN'},
  {'geo_name':'Brazil', 'geo_code':'BRA'},
  {'geo_name':'Low Income Countries', 'geo_code':'LIC'},
  {'geo_name':'Sub-Saharan Africa', 'geo_code':'SSF'},
  {'geo_name':'OECD', 'geo_code': 'OEC'}
  ]

geographies_df = pd.DataFrame(geographies_dict)
for idx, row in geographies_df.iterrows():
  display(Markdown(f"#### Dependency Ratio for {row['geo_name']}"))
  data = fred.get_series(f"SPPOPDPNDOL{row['geo_code']}", observation_start='1960-01-01', observation_end='2020-01-01')
  df = pd.DataFrame(data)
  df.plot()
# I first get all text, then all images
# how to put the figure identifier?

```
Am I doing something obviously wrong?

**Comment by jjallaire** [comment, marked as answer]
Not sure exactly what df.plot is doing, but this example using `plot.show()` works the way you are expecting:

````python
```{python}
import matplotlib.pyplot as plt
from IPython.display import display, Markdown

for x in range(1, 4):
  display(Markdown(f"## Plot {x}"))
  plt.plot([1,23,2,4])
  plt.show()
```
````

`df.plot()` is likely just "attaching" a plot to the cell rather than creating a plot within the stream of outputs.

**Reply by lucas-a-meyer** [reply]
Thank you so much @jjallaire - this worked perfectly.
